packages/ingest/src/FtpProviderClient.ts

```diff
@@ -1,63 +1,96 @@
-'use strict';
-
-const get = require('lodash/get');
-const JSFtp = require('jsftp');
-const { PassThrough } = require('stream');
-const log = require('@cumulus/common/log');
-const omit = require('lodash/omit');
-const S3 = require('@cumulus/aws-client/S3');
-const isNil = require('lodash/isNil');
-const { recursion } = require('./recursion');
-const { lookupMimeType, decrypt } = require('./util');
+import JSFtp, { ListError } from 'jsftp';
+import { PassThrough } from 'stream';
+import log from '@cumulus/common/log';
+import S3 from '@cumulus/aws-client/S3';
+import isNil from 'lodash/isNil';
+import { Socket } from 'net';
+import { recursion } from './recursion';
+import { lookupMimeType, decrypt } from './util';
+import { FtpProviderClientListItem, ProviderClientListItem } from './types';
+
+interface FtpProviderClientConstructorParams {
+  host: string;
+  port?: number;
+  useList?: boolean;
+  username?: string;
+  password?: string;
+  encrypted?: boolean;
+}
+
+function isJSFtpError(error: Error | ListError): error is ListError {
```

Comment:
What does the `error is ListError` part of this do? Specify that is what the return should be? Because we're obviously testing to determine if this error is a ListError?

Reply:
It's a [User-Defined Type Guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards), which tells TypeScript what type an object is.